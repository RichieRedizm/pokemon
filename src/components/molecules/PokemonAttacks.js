import React from 'react'
const { formatObjectList } = require('../helper')

const PokemonAttacks = ({ attacks }) => {
  return (
    <div data-testid='pokemon-attacks' className='card detail'>
      {attacks && <h3 className='label-dark'>Attacks</h3>}
      {attacks.fast && (
        <div>
          <strong className='label-dark'>Fast Attacks: </strong>
          {formatObjectList(attacks.fast)}
        </div>
      )}
      {attacks.special && (
        <div>
          <strong className='label-dark'>Special Attacks: </strong>
          {formatObjectList(attacks.special)}
        </div>
      )}
    </div>
  )
}

export default PokemonAttacks
