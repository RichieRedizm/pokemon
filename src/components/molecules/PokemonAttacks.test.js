import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import PokemonAttacks from './PokemonAttacks'

test('it displays a pokemon list item', () => {
  const pokemonAttacks = {
    fast: [
      {
        name: 'Tackle',
        type: 'Normal',
        damage: 12
      },
      {
        name: 'Vine Whip',
        type: 'Grass',
        damage: 7
      }
    ],
    special: [
      {
        name: 'Power Whip',
        type: 'Grass',
        damage: 70
      },
      {
        name: 'Seed Bomb',
        type: 'Grass',
        damage: 40
      },
      {
        name: 'Sludge Bomb',
        type: 'Poison',
        damage: 55
      }
    ]
  }
  const { getByTestId, asFragment } = render(
    <Router>
      <PokemonAttacks attacks={pokemonAttacks} />
    </Router>
  )

  expect(getByTestId('pokemon-attacks'))
  expect(asFragment()).toMatchSnapshot()
})
