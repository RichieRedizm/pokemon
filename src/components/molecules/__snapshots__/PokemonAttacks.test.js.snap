// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays a pokemon list item 1`] = `
<DocumentFragment>
  <div
    class="card detail"
    data-testid="pokemon-attacks"
  >
    <h3
      class="label-dark"
    >
      Attacks
    </h3>
    <div>
      <strong
        class="label-dark"
      >
        Fast Attacks: 
      </strong>
      <div
        class="detail-object-list"
      >
        <span>
          <strong
            class="label-light"
          >
            Name:
          </strong>
           Tackle
        </span>
        <span>
          <strong
            class="label-light"
          >
            Type:
          </strong>
           Normal
        </span>
        <span>
          <strong
            class="label-light"
          >
            Damage:
          </strong>
           12
        </span>
      </div>
      <div
        class="detail-object-list"
      >
        <span>
          <strong
            class="label-light"
          >
            Name:
          </strong>
           Vine Whip
        </span>
        <span>
          <strong
            class="label-light"
          >
            Type:
          </strong>
           Grass
        </span>
        <span>
          <strong
            class="label-light"
          >
            Damage:
          </strong>
           7
        </span>
      </div>
    </div>
    <div>
      <strong
        class="label-dark"
      >
        Special Attacks: 
      </strong>
      <div
        class="detail-object-list"
      >
        <span>
          <strong
            class="label-light"
          >
            Name:
          </strong>
           Power Whip
        </span>
        <span>
          <strong
            class="label-light"
          >
            Type:
          </strong>
           Grass
        </span>
        <span>
          <strong
            class="label-light"
          >
            Damage:
          </strong>
           70
        </span>
      </div>
      <div
        class="detail-object-list"
      >
        <span>
          <strong
            class="label-light"
          >
            Name:
          </strong>
           Seed Bomb
        </span>
        <span>
          <strong
            class="label-light"
          >
            Type:
          </strong>
           Grass
        </span>
        <span>
          <strong
            class="label-light"
          >
            Damage:
          </strong>
           40
        </span>
      </div>
      <div
        class="detail-object-list"
      >
        <span>
          <strong
            class="label-light"
          >
            Name:
          </strong>
           Sludge Bomb
        </span>
        <span>
          <strong
            class="label-light"
          >
            Type:
          </strong>
           Poison
        </span>
        <span>
          <strong
            class="label-light"
          >
            Damage:
          </strong>
           55
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
